### PR TITLE
feat: adapt to nextjs 9.4.5

### PR DESCRIPTION
**Pull request by migatech2015**
This PR should solve #88 since it adapts this module to the Next.js's webpack config in version 9.4.5

### CODE REVIEW

**Comment by martpie** [comment]
Thank you very much for the PR, this is unfortunately a breaking change so we'll have to bump the major version. 

I would like to wait a little bit before merging as the next version of Next.js may be a major one and is not out yet.

ps: you have failing tests ;)

File: src/next-transpile-modules.js [hunk 1]
```diff
@@ -99,13 +99,13 @@ const withTmInitializer = (transpileModules = []) => {
...
-            nextCssLoader.issuer.include = nextCssLoader.issuer.include.concat(includes);
-            nextCssLoader.issuer.exclude = excludes;
+            nextCssLoader.issuer.and = nextCssLoader.issuer.and.concat(includes);
+            nextCssLoader.issuer.not = excludes;
           }
 
           if (nextSassLoader) {
-            nextSassLoader.issuer.include = nextCssLoader.issuer.include.concat(includes);
-            nextSassLoader.issuer.exclude = excludes;
+            nextSassLoader.issuer.and = nextSassLoader.issuer.and.concat(includes);
```
Comment (dan-wu39) [inline src/next-transpile-modules.js:107]: :o I'm surprised `nextCssLoader` being used instead of `nextSassLoader ` wasn't causing issues before...

Comment (martpie) [reply src/next-transpile-modules.js:107]: 🙈 apparently it worked fine

Comment (migatech2015) [reply src/next-transpile-modules.js:107]: luckily they had the same includes that's why it didn't cause any problems 😄

**Comment by migatech2015** [comment]
@martpie yes I'll have a look at the tests today. Should I increase the version number or do you want to do that? We could also keep it backwards compatible and check if `issuer.include` or `issuer.and` is set.

**Comment by migatech2015** [comment]
I've updated the nextjs version used by the tests but they run into errors when importing a sass or scss file. Does someone know why? I compared the webpack config generated by nextjs 9.4.4 with the current one and there seem to be no changes concerning the sass-loader.

**Comment by dan-wu39** [comment]
You're talking about this error? Just a guess, but are you able to see that the `node_modules/shared-ui` folder is included in the sassLoader.and array?

```
./node_modules/shared-ui/components/Input.module.scss 1:12
Module parse failed: Identifier directly after number (1:12)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> $padding: 10px;
| $bg: black;
| $color: white;
```

**Comment by migatech2015** [comment]
@dan-wu39 yes that's the error I mean and yes the `node_modules/shared-ui` folder gets appended to `sassLoader.issuer.and` like before the nextjs upgrade. I didn't find a hint so far that the sass loader won't compile files from `node_modules` anylonger and nextjs 9.4.4 and 9.4.5 uses the same version for `sass-loader` 8.0.2

**Comment by alejandronanez** [comment]
👋 Thanks @migatech2015 for working on this 🚀

I'm really interested in seeing this shipped, is there anything you need help with? happy to give you a hand if needed! cc @martpie 

Thanks again!

**Comment by martpie** [comment]
@alejandronanez I personally don't have a lot of incentive on working on this PR specifically, as I recommend using stable versions of Next.js. Merging this also comes with a couple of problems to solve (but I can handle them myself, it's fine)

If the tests get fixed and pass correctly, I'm ok with merging it and releasing a pre-release version of the plugin if it can be helpful to some people, but I don't want to spend too much time yet on writing changelogs/compatibility tables etc as long as we don't know if the next version of Next.js will be a major or minor (most probably major).

Hope you understand :) 👍

td,dr: I usually only work on the plugin when a new version of Next.js comes out 😄

**Comment by ratheo** [comment]
`include.and` means that all conditions must match. Instead you should use `include.or` (https://webpack.js.org/configuration/module/#condition)

The 'Hack our way to disable errors on node_modules CSS modules' hack should at least use the same logic as next does, in order for it to work in both an interactive and non interactive shell. (https://github.com/vercel/next.js/blob/canary/packages/next/build/webpack/config/blocks/css/messages.ts)

**Comment by alejandronanez** [comment]
> td,dr: I usually only work on the plugin when a new version of Next.js comes out 😄
> 
> @alejandronanez I personally don't have a lot of incentive on working on this PR specifically, as I recommend using stable versions of Next.js. Merging this also comes with a couple of problems to solve (but I can handle them myself, it's fine)
> 
> If the tests get fixed and pass correctly, I'm ok with merging it and releasing a pre-release version of the plugin if it can be helpful to some people, but I don't want to spend too much time yet on writing changelogs/compatibility tables etc as long as we don't know if the next version of Next.js will be a major or minor (most probably major).
> 
> Hope you understand :) 👍

@martpie yes, I totally get it and makes complete sense to me.

I think I'd wait for this to be released once Next.js publishes its new version.

Thank you again @martpie! 👍

**Comment by migatech2015** [comment]
I'm using `issuer.or` now instead of `issuer.and` and this seems to slove our problem and fix the tests 🎉 
Thank you @ratheo for helping me out and I've also added the error messages for interactive and non-interactive shells so that the exclude should work with either of these.

**Pull request merged**